src/wifi/mod.rs

```diff
@@ -1,43 +1,149 @@
+    pub flags NM80211ApFlags: u32 {
+        // access point has no special capabilities
+        const AP_FLAGS_NONE           = 0x00000000,
+        // access point requires authentication and
+        const AP_FLAGS_PRIVACY        = 0x00000001,
+    }
+}
+
+
+bitflags! {
+    pub flags NM80211ApSecurityFlags: u32 {
+         // the access point has no special security requirements
+        const AP_SEC_NONE                    = 0x00000000,
+        // 40/64-bit WEP is supported for pairwise/unicast encryption
+        const AP_SEC_PAIR_WEP40              = 0x00000001,
+        // 104/128-bit WEP is supported for pairwise/unicast encryption
+        const AP_SEC_PAIR_WEP104             = 0x00000002,
+        // TKIP is supported for pairwise/unicast encryption
+        const AP_SEC_PAIR_TKIP               = 0x00000004,
+        // AES/CCMP is supported for pairwise/unicast encryption
+        const AP_SEC_PAIR_CCMP               = 0x00000008,
+        // 40/64-bit WEP is supported for group/broadcast encryption
+        const AP_SEC_GROUP_WEP40             = 0x00000010,
+        // 104/128-bit WEP is supported for group/broadcast encryption
+        const AP_SEC_GROUP_WEP104            = 0x00000020,
+        // TKIP is supported for group/broadcast encryption
+        const AP_SEC_GROUP_TKIP              = 0x00000040,
+        // AES/CCMP is supported for group/broadcast encryption
+        const AP_SEC_GROUP_CCMP              = 0x00000080,
+        // WPA/RSN Pre-Shared Key encryption is supported
+        const AP_SEC_KEY_MGMT_PSK            = 0x00000100,
+        // 802.1x authentication and key management is supported
+        const AP_SEC_KEY_MGMT_802_1X         = 0x00000200,
+    }
+}
 
 
 /// Scans for Wi-Fi access points.
 ///
 /// # Examples
 ///
 /// ```
-/// let access_points = network_manager::wifi::scan().unwrap();
+/// use network_manager::manager;
+/// use network_manager::wifi;
+/// use network_manager::device;
+/// let manager = manager::new();
+/// let mut devices = device::list(&manager).unwrap();
+/// let i = devices.iter().position(|ref d| d.device_type == device::DeviceType::WiFi).unwrap();
+/// let device = &mut devices[i];
+/// let access_points = wifi::scan(&manager, device).unwrap();
 /// println!("{:?}", access_points);
 /// ```
-pub fn scan() -> Result<Vec<AccessPoint>, String> {
-    // Scan for access points
-
-    let ap1 = AccessPoint {
-        ssid: "ap1".to_string(),
-        signal: 60,
-        security: vec![Security::WEP],
-    };
-
-    let ap2 = AccessPoint {
-        ssid: "ap2".to_string(),
-        signal: 92,
-        security: vec![Security::WPA1, Security::WPA2],
-    };
-
-    Ok(vec![ap1, ap2])
+pub fn scan(manager: &NetworkManager, device: &Device) -> Result<Vec<AccessPoint>, String> {
+    let mut access_points = Vec::new();
+
+    if device.device_type == device::DeviceType::WiFi {
+        let paths = try!(manager.get_device_access_points(&device.path));
+
+        for path in paths {
```

Comment:
I just checked and there is a better fit with the `filter_map` method, but I could not get it to work because of the `try!`.